junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertTimeout.java

```diff
@@ -124,7 +126,12 @@ static void assertTimeoutPreemptively(Duration timeout, Executable executable, S
 	private static <T> T assertTimeoutPreemptively(Duration timeout, ThrowingSupplier<T> supplier,
 			Object messageOrSupplier) {
 
-		ExecutorService executorService = Executors.newSingleThreadExecutor();
+		AtomicReference<Thread> threadReference = new AtomicReference<>();
+		ExecutorService executorService = Executors.newSingleThreadExecutor(runnable -> {
+			Thread thread = Executors.defaultThreadFactory().newThread(runnable);
+			threadReference.set(thread);
+			return thread;
+		});
```

Comment:
Do we need the special `ThreadFactory`? Couldn't we just set the `threadReference` in the `Runnable` passed to `submit` below?

Reply:
Good one. 👌